### ISSUE TITLE
engine.save_to_file() produces empty file

### ISSUE THREAD

**Issue by angoodkind** [nateshmbhat/pyttsx3#142]
In running the code below, `engine.say()` works fine. However, when I save the audio to a file, the file is empty. I have also verified this in `sox`.

I am running `pyttsx3` on a Macbook running OS 10.15.6.
```
import pyttsx3
engine = pyttsx3.init()
engine.say("I will speak this text") # successfully produced
engine.setProperty('volume',1.0)
volume = engine.getProperty('volume') 
print(volume)  # prints 1
engine.save_to_file('Hello World', 'test.mp3')
engine.runAndWait()
# engine.stop()
```

**Comment by akkapakasaikiran** [comment]
`save_to_file` seems to work in interactive mode but not in scripting mode.

```
$ python3
>>> import pyttsx3; engine = pyttsx3.init(); engine.save_to_file('Hello world', 'tmp.mp3'); engine.runAndWait();
```
Works as expected. But the following two ways don't work (they don't even create an empty file) (I am using Ubuntu 18.04).
``` 
$ python3 -c "import pyttsx3; engine = pyttsx3.init(); engine.save_to_file('Hello world', 'tmp.mp3'); engine.runAndWait();" 
$ python3 main.py 
``` 

where `main.py` is as follows:
```
import pyttsx3
engine = pyttsx3.init()
engine.save_to_file('Hello world', 'tmp.mp3')
engine.runAndWait()
``` 

Note that `engine.say()` is working (for me) in all three ways of invoking python. 

Looking forward to any updates.

**Comment by shakhrom96bs** [comment]
Hello I have some issue with this code 



import pyttsx3
engine = pyttsx3.init()
engine.save_to_file('Hello World' , 'test.mp3')
engine.runAndWait()

when I execute this Code. I get : AttributeError: 'Engine' object has no attribute 'save_to_file'

**Comment by shakhrom96bs** [comment]
can anyboda help me.

**Comment by shakhrom96bs** [comment]
y*

**Comment by awesome-amy** [comment]
I also got empty file when saving to ".mp3", but simply changing to ".aiff"  resolved my issue. 

According to Apple's documentation, NSSpeechSynthesizer synthesizes text into a sound (AIFF) file, so ".aiff" should be the supported format. (macOS 10.3+)
[https://developer.apple.com/documentation/appkit/nsspeechsynthesizer/1448394-startspeakingstring?language=objc](url)

**Comment by bqtd** [comment]
what about windows? I did the same thing with windows and it just saved nothing

**Comment by MoYuFly** [comment]
me too，but in myself computer Normal operation ，Cloud computer make empty file，the cloud computer does not have a graphics card installed。Windows 10 system installed on cloud computer，can you help me.

**Comment by jgframoz** [comment]
> I also got empty file when saving to ".mp3", but simply changing to ".aiff" resolved my issue.
> 
> According to Apple's documentation, NSSpeechSynthesizer synthesizes text into a sound (AIFF) file, so ".aiff" should be the supported format. (macOS 10.3+) [https://developer.apple.com/documentation/appkit/nsspeechsynthesizer/1448394-startspeakingstring?language=objc](url)

That solved my problem! Thanks!

**Comment by gustavz** [comment]
For me (on macOS) it is able to create / save a .`wav` or `.aiff` file, but these are not readable / playable

**Comment by Mateja9991** [comment]
> For me (on macOS) it is able to create / save a .`wav` or `.aiff` file, but these are not readable / playable

I have the same problem, did you manage to fix it?

**Comment by IlyaYakushevskiy** [comment]
> For me (on macOS) it is able to create / save a .`wav` or `.aiff` file, but these are not readable / playable

The same problem with mac, works on windows tho. Did you solve it?

**Comment by bearcatjamboree** [comment]
This still produces empty files on mac, whether you use .wav, .mp3, or .aiff.  Anyone know a way to fix this?

**Comment by egorgam** [comment]
I found a small hack to fix it https://github.com/nateshmbhat/pyttsx3/pull/267. Maybe someone can found better solution to fix it right 🥲

**Comment by nateshmbhat** [comment]
fixed in #271

**Comment by varnav** [comment]
CentOS 7 (yeah, I know, it's old)

This code

```
import pyttsx3
engine = pyttsx3.init() # object creation
engine.save_to_file('Hello World', 'test.mp3')
engine.runAndWait()
```

or this code (wav)

```
import pyttsx3
engine = pyttsx3.init() # object creation
engine.save_to_file('Hello World', 'test.wav')
engine.runAndWait()
```

Produce empty files for me. They have relevant headers for mp3 file or wav file, but nothing else.
